okio/jvm/src/main/java/okio/-Unsigned.kt

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("-Unsigned")
+@file:Suppress("NOTHING_TO_INLINE") // Syntactic sugar.
+@file:UseExperimental(ExperimentalUnsignedTypes::class)
+
+package okio
+
+import java.io.IOException
+
+/** Marker interface for APIs related to unsigned numbers. */
+@Experimental
+annotation class OkioUnsignedApi
+
+// BufferedSink
+
+/** Writes a unsigned byte to this sink. */
+@OkioUnsignedApi
+@Throws(IOException::class)
+inline fun <S : BufferedSink> S.writeUByte(b: UByte): S {
+  writeByte(b.toByte().toInt())
+  return this
+}
+
+/**
+ * Writes a big-endian unsigned short to this sink using two bytes.
```

Comment:
Grammar: preference on "big-endian, unsigned short" or "unsigned, big-endian short"?

Reply:
I think “unsigned short” is the subject, and big-endian is a representation of that subject . . . 

so I like big-endian, unsigned short